test/integration_test.go

```diff
@@ -197,6 +199,82 @@ const (
 	fixtureDownload = "fixture-download"
 )
 
+func TestTerragruntProviderCache(t *testing.T) {
```

Comment:
Thanks for adding this test! 

Given the rather large impact this has across all TG functionality, could we find a way to run _all_ TG integration tests with (a) caching enabled and (b) caching disabled? I want to be sure all existing functionality still works with caching.

Reply:
Added tasks with provider cache.